exp/addons/internal/controllers/clusterresourceset_helpers.go

```diff
@@ -196,3 +191,19 @@ func computeHash(dataArr [][]byte) string {
 	}
 	return fmt.Sprintf("sha256:%x", hash.Sum(nil))
 }
+
+func getClusterNameFromOwnerRef(obj metav1.ObjectMeta) (string, error) {
+	for _, ref := range obj.OwnerReferences {
+		if ref.Kind != "Cluster" {
+			continue
+		}
+		gv, err := schema.ParseGroupVersion(ref.APIVersion)
+		if err != nil {
+			return "", errors.WithStack(err)
+		}
+		if gv.Group == clusterv1.GroupVersion.Group {
```

Comment:
Let's additionally return an error here if the name is empty - saves from checking that case where this is called, and the error should never occur.